oligos/models.py

```diff
@@ -34,13 +43,354 @@ class Oligo(models.Model):
     grade = models.CharField(max_length=80, blank=True)
 
     def __str__(self):
+        """String represnting the Oligo object"""
         return self.name
 
+    def get_absolute_url(self):
+        """Returns the url to access a particular instance of the model."""
+        return reverse('detail', args[str(self.id)])
+
     def length(self):
         return len(self.sequence)
 
-    def CanEdit(self, request):
-        return (request.user == self.user)
+    class Meta:
+        ordering = ['create_date']
+
+
+def extract_number(oligoname):
+    match = re.search('\d\d\d\d\d*',oligoname)
+    if match:
+        return int(match.group(0))
+    else:
+        return None
+
+def auto_name():
+    q = Oligo.objects.all().order_by('-create_date')[:100]
+    namelist = list(q.values_list('name', flat=True))
+    numlist = []
+    for name in namelist:
+        num = extract_number(name)
+        if num:
+            numlist.append(num)
+    numlist.sort(reverse=True)
+    topnum = numlist[0]
+    newname = 'RU-O-'+str(topnum+1)
+    return newname
+
+def auto_name_oligolist(oligolist):
+    for oligo in oligolist:
+        oligo.name = auto_name()
+        oligo.save()
+    return oligolist
+
+
+class UserMap(models.Model):
```

Comment:
I think we can completely delete the UserMap

Reply:
Deleted!